File: Array/RemoveElement.cpp
#include <iostream>
using namespace std;
class Solution{
    public:
        int removeElement(int A[], int n, int elem){
            int i = 0;
            int j = 0;
            for(i = 0; i < n; i++){
                printf("%d\n",A[i]);
                if(A[i] == elem){
                    continue;
                }

                A[j] = A[i];
                j++;
            }
            return j;
        }
};
int main(){
    Solution solution;
    int A[7] = {1,2,3,4,5,6,4};
    int length = solution.removeElement(A,7,4);
    printf("%d\n",length);
};

File: remove-element.py
from typing import List


class Solution:
    # 方法1
    def removeElement(self, nums: List[int], val: int) -> int:
        for i in range(len(nums)-1, -1, -1):
            if nums[i] == val:
                nums.pop(i)
        return len(nums)


if __name__ == '__main__':
    s = Solution()
    print(s.removeElement([3, 2, 2, 3], 3))
    print(s.removeElement([0, 1, 2, 2, 3, 0, 4, 2], 2))
